drivers/wifi/esp32/Kconfig.esp32

```diff
@@ -0,0 +1,226 @@
+# Copyright 2020 Espressif Systems (Shanghai) PTE LTD
+
+menuconfig WIFI_ESP32
+	bool "ESP32 SoC WiFi support"
+	help
+	  Enable ESP32 SoC WiFi support
+
+if WIFI_ESP32
+
+config WIFI_SSID
+	string "SSID of WiFi network"
+	help
+	  SSID (network name) for the application to connect to
+
+config WIFI_PASSWORD
+	string "Password (WPA or WPA2) of WiFi network"
+	help
+	  WiFi password (WPA or WPA2) for the example to use.
+
+config WIFI_STA_AUTO
+	bool "Automatically connect to configured WiFi SSID"
+	help
+	  WiFi driver will automatically connect to SSID
+
+config WIFI_STA_RECONNECT
+	bool "WiFi connection retry"
+	default y
+	help
+	  Set auto WiFI reconnection when disconnected
+
+config EVENT_TASK_STACK_SIZE
+	int "Event Task Stack Size"
+	default 4096
+
+config EVENT_TASK_PRIO
+	int "Event Task Priority"
+	default 4
```

Comment:
Initial implementation had "default" values. However, I took @jukkar comment (https://github.com/zephyrproject-rtos/zephyr/pull/32081#discussion_r572107170) and removed that. Now it is up to the user to set in `prj.conf` file.  
Regardinf the prefix, I'll set `ESP32_WIFI` as default to meet all other already implement options.

Reply:
I meant to change `config WIFI_PASSWORD` into `config WIFI_ESP32_DEFAULT_PASSWORD`, because I thought that this is the default password that can be changed using wifi mgmt API. But as I can see, Kconfig is the only place so far that allows to configure ssid/psk... so ignore my suggestion with "DEFAULT". I guess that the next goal after this PR is to integrate with wifi mgmt API and drop those options from Kconfig, right?

Reply:
Yes, you are correct. Anyway, I'll only modify the options adding `ESP32_WIFI` to all instances.